src/platforms/node/services/node-environment-service.ts

```diff
@@ -0,0 +1,59 @@
+import { Level } from '../../../level'
+import { Environment } from '../../../environment'
+
+export class NodeEnvironmentService {
```

Comment:
`NodeEnvironmentService` and `WebEnvironmentService` are 99% duplicates of one another. A base abstract class for them should be pretty easy to extract.

In total there are only 2 lines that differ inside the class. One is already contained within the private method (`getRawLevel()`). The other is when getting the `rawEnvironment`, which then should just be extracted to a similar method like `getRawLevel()`.

Reply:
I have extracted the common logic into the class `EnvironmentService` and moved the `rawEnvironment` into its own function.

Reply:
Hmm. That helped, but there are still alot of duplication.
What i had in mind was for something along the lines of the below code.

``` typescript
export abstract class EnvironmentService {  
  //Insert all the other code that was extracted

  public getLevel(): Level {
    const environment = this.getEnvironment()
    const fallbackLevel = this.environmentService.getEnvironmentLevel(environment)
    const rawLevel = this.getRawLevel()
    return this.sanitizeLevel(rawLevel, fallbackLevel)
  }

  public getEnvironment(): Environment {
    const rawEnvironment = this.getRawEnvironment()
    return this.sanitizeEnvironment(rawEnvironment)
  }

  protected abstract getRawEnvironment(): string 
  protected abstract getRawLevel(): string 
}

export class NodeEnvironmentService extends EnvironmentService {
  protected getRawEnvironment(): string {
    return process.env.NODE_ENV?.toLowerCase() ?? ''
  }
  protected getRawLevel(): string {
    return process.env.LOG_LEVEL?.toLowerCase() ?? ''
  }
}

export class WebEnvironmentService extends EnvironmentService {
  protected getRawEnvironment(): string {
    return window.env?.ENV?.toLowerCase() ?? ''
  }
  protected getRawLevel(): string {
    return window.env?.LOG_LEVEL?.toLowerCase() ?? ''
  }
}
```

With the above shown structure, there would be no duplication, which isn't necessary for some reason, like e.g the implemented abstract methods names.

Unsure if the code compiles/works in typescript, but it should from my understanding.